src/RoslynPad.Editor.Windows/SearchReplacePanel.cs

```diff
@@ -711,7 +725,7 @@ public void Draw(TextView textView, DrawingContext drawingContext)
             var viewStart = visualLines.First().FirstDocumentLine.Offset;
             var viewEnd = visualLines.Last().LastDocumentLine.EndOffset;
 
-            foreach (var result in CurrentResults.FindOverlappingSegments(viewStart, viewEnd - viewStart))
+            foreach (var result in CurrentResults.Where(r => viewStart <= r.Offset && r.Offset <= viewEnd || viewStart <= r.EndOffset && r.EndOffset <= viewEnd))
```

Comment:
Are you sure this `Where` covers all the functionality in `FindOverlappingSegments`? The logic there seems a lot more complex.

Reply:
I haven't tried to cover all the functionality in `FindOverlappingSegments`, only provide the behaviour that's needed by the `Draw` method. That method appeared to use `FindOverlappingSegments` to get all the segments that were currently visible so I've tried to provide an implementation that achieves the same thing.

I think it'd be worth having unit tests for this behaviour if you don't mind me making the `SearchReplaceResultBackgroundRenderer` public so it can be tested?

Having said that, I've taken a quick look at `FindOverlappingSegments` in the current AvalonEdit codebase. I'm hoping it just looks complicated because it's traversing a binary tree structure in `TextSegmentCollection`. I'll need to debug the code to properly understand it though. I'll do that and see if I can find anything we ought to worry about..

Reply:
I've taken a closer look at FindOverlappingSegments and it looks to be as I thought - finding all the segments that overlap a certain offset range. So I'm satisfied I've captured the behaviour that's needed here.

Hopefully I'll find time to try add unit tests for that and convert all the unit tests to XUnit in the next few days.

Reply:
Hi @aelij 

I've tried to add unit tests for `SearchReplaceResultBackgroundRenderer.Draw` (where I've replaced the call to FindOverlappingSegments). Unfortunately the `Draw` method takes a `TextView` (from AvalonEdit) as an argument and this makes writing decent unit tests quite difficult - I'm unable to mock the argument and manipulating it into the state it would need to be means writing tests that rely quite heavily on the implementation of `TextView`. So I've decided not to introduce the unit tests.

I have converted the unit tests from NUnit to XUnit though. This relies on an unofficial NuGet though (https://github.com/AArnott/Xunit.StaFact) to run the tests on the right thread context. I'm a little concerned about this - there could be a good reason that this isn't supported natively by XUnit and there's not much guarantee that the NuGet will continue to be supported. 

My advice would be to use NUnit but of course it's your project and you might know a good reason to prefer XUnit that I don't.

So I think the PR is ready for you to take another look.